dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/operators/EqualOperator.java

```diff
@@ -114,44 +114,44 @@ public boolean test( Object value )
             String s1 = getValue( String.class );
             String s2 = (String) value;
 
-            return s1 != null && s2.equals( s1 );
+            return s2.equals( s1 );
         }
-        else if ( type.isBoolean() )
+        if ( type.isBoolean() )
         {
             Boolean s1 = getValue( Boolean.class );
             Boolean s2 = (Boolean) value;
 
-            return s1 != null && s2.equals( s1 );
+            return s2.equals( s1 );
```

Comment:
Is doing multiple IFs better than else ifs?

Reply:
I think it better reflects the control flow that is actually happening and there is a warning in the IDE so I might not be alone with this opinion. The machine code will be the same.